include/pm/device.h

```diff
@@ -424,8 +431,59 @@ static inline bool pm_device_wakeup_is_capable(const struct device *dev)
 {
 	return false;
 }
+
 #endif /* CONFIG_PM_DEVICE */
 
+#ifdef CONFIG_PM_DEVICE_CONSTRAINT
+/**
+ * @brief Set a constraint on the current device state.
+ *
+ * This function sets a constraint in the current device state. After the
+ * constraint is set the system power management and device runtime power
+ * management will not longer change the device state until the constraint be
+ * released.
+ *
+ * @see pm_device_constraint_release
+ *
+ * @param dev Device instance.
+ */
+void pm_device_constraint_set(struct device *dev);
```

Comment:
note that "constraint" is a really generic word, here it means that device state is _locked_, so I'd choose a better API naming.

Reply:
agree, I just don't know if locked is much better, I was thinking that enable / disable was more appropriate but then it will be confuse with other APIs.

Reply:
hmm, `pm_device_state_lock()`, `pm_device_state_unlock()` and `pm_device_state_is_locked()` sounds good though.